src/Xamarin.Android.Build.Tasks/Utilities/NdkTools/NdkToolKind.cs

```diff
@@ -0,0 +1,11 @@
+namespace Xamarin.Android.Tasks
+{
+	public enum NdkToolKind
+	{
+		Assembler,
+		CompilerC,
```

Comment:
Given issues reported "elsewhere" (don't want to find a ref) wherein on Windows the `clang.cmd` script actually invokes `clang++.exe`, is there some way that we could "just" deal with C++ code and not C code?  In effect, *not have* an `NdkToolKind.CompilerC` enum value, and just use `NdkToolKind.CompilerCPlusPlus` everywhere?

Reply:
`mkbundle` requires a C compiler (or we'd have to change it to pass `-x C` to the C++ compiler). And the `clang.cmd` invokes clang++ only in NDK 19 and 19b, 19c fixes it:

```cmd
@echo off
setlocal
call :find_bin
if "%1" == "-cc1" goto :L

set "_BIN_DIR=" && %_BIN_DIR%clang.exe --target=armv7a-linux-androideabi16 -fno-addrsig %*
if ERRORLEVEL 1 exit /b 1
goto :done

:L
rem Target is already an argument.
set "_BIN_DIR=" && %_BIN_DIR%clang.exe %*
if ERRORLEVEL 1 exit /b 1
goto :done

:find_bin
rem Accommodate a quoted arg0, e.g.: "clang"
rem https://github.com/android-ndk/ndk/issues/616
set _BIN_DIR=%~dp0
exit /b

:done
```

We already hack around the above CMD (due to quoting issues), I don't want to introduce another hack to make `mkbundle` work with a C++ compiler :)

Reply:
…Meaning this is a place that we can cleanup when we drop support for the Legacy SDK, and thus also drop support for `mkbundle` entirely.

Reply:
When that happens we can drop the compiler support completely, as AOT (and us, as well) use only the assembler, the linker and strip